### PR TITLE
(1.x) STORM-2555 handle impersonation properly for HBase delegation token

### PR DESCRIPTION
**Pull request by HeartSaVioR**
Please refer [STORM-2555](http://issues.apache.org/jira/browse/STORM-2555) for more details.

There had been code block for handling impersonation, but actually it didn't work.

Please review and comment, thanks in advance.

PR for master branch: #2160

### CODE REVIEW

File: external/storm-autocreds/src/main/java/org/apache/storm/hbase/security/HBaseSecurityUtil.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public static UserProvider login(Map conf, Configuration hbaseConfig) throws IOE
             }
             return legacyProvider;
         } else {
-            return UserProvider.instantiate(hbaseConfig);
+            return null;
```
Comment (arunmahadevan) [inline external/storm-autocreds/src/main/java/org/apache/storm/hbase/security/HBaseSecurityUtil.java]: Wont this throw NPE in the caller ?

check: https://github.com/apache/storm/blob/1.x-branch/external/storm-hbase/src/main/java/org/apache/storm/hbase/trident/state/HBaseMapState.java#L84

Comment (HeartSaVioR) [inline external/storm-autocreds/src/main/java/org/apache/storm/hbase/security/HBaseSecurityUtil.java]: I think I just completely missed HBaseMapState. Will fix.

Comment (HeartSaVioR) [inline external/storm-autocreds/src/main/java/org/apache/storm/hbase/security/HBaseSecurityUtil.java]: Addressed.

**Comment by HeartSaVioR** [comment]
@arunmahadevan 
Thanks for reviewing. I addressed your review comment. 
Could you please take a look again? Thanks!

**Comment by arunmahadevan** [comment]
+1

**Pull request merged**